crates/driver/src/infra/simulator/mod.rs

```diff
@@ -180,7 +182,7 @@ where
             {
                 Some(tx)
             }
```

Comment:
This special case is not identical to the general case right below.

Reply:
right, reverted